### PR TITLE
update tests

### PR DESCRIPTION
**Pull request by ermalkaleci**
* Update tests & snapshots
* Add test helper to make it easy to write e2e tests
* Add a test example

### CODE REVIEW

File: tests/xcm-transfer/shared.ts [hunk 1]
```diff
@@ -19,6 +18,27 @@ type TestType =
   | PolkadotParaTestType
   | PlaygroundTestType
 
+// TODO: figure out whey using chopsticks-testing impl will not find 'XcmVersionedXcm'
```
Comment (shaunxw) [inline tests/xcm-transfer/shared.ts:21]: Is this for future tasks?

File: tests/xcm-transfer/shared.ts [hunk 2]
```diff
@@ -31,18 +51,18 @@ export default function buildTest(tests: ReadonlyArray<TestType>) {
...
-        toAccount = opt.toAccount(ctx)
+        toAccount = (opt.toAccount as any)(ctx)
       }
 
-      let precision = 3
-      if ('precision' in opt) {
-        precision = opt.precision
-      }
+      const precision = 3
+      // if ('precision' in opt) {
```
Comment (shaunxw) [inline tests/xcm-transfer/shared.ts:63]: why deprecate the `precision` option?

File: tests/xcm-transfer/shared.ts (hunk 1 above)
Comment (ermalkaleci) [reply tests/xcm-transfer/shared.ts:21]: don't know. for some reason current runtime doesn't work. Tried upcoming runtime and it works. So this is tmp approach

File: tests/xcm-transfer/shared.ts (hunk 2 above)
Comment (ermalkaleci) [reply tests/xcm-transfer/shared.ts:63]: sync with acala's e2e tests. I plan to replace this completely

Comment (ermalkaleci) [reply tests/xcm-transfer/shared.ts:63]: current setup is very hard to understand and keep up to date. every snapshot goes into one file. making it hard to understand which test failed, why failed, and why snapshot was update.
I plan to migrate every test into its file and write it like this https://github.com/AstarNetwork/e2e-tests/pull/26/files#diff-ede7f10848ecdee45d03f542c782549b4af14f83511e8cac1ec47fc41702a240

File: tests/xcm-transfer/shared.ts (hunk 1 above)
Comment (shaunxw) [reply tests/xcm-transfer/shared.ts:21]: ok, let me create an issue for it. could be resolved later

**Review by shaunxw — APPROVED** [review]

**Pull request merged**